proxy/delete.go

```diff
@@ -61,17 +61,19 @@ func DeleteFunctionToken(gateway string, functionName string, tlsInsecure bool,
 	case http.StatusOK, http.StatusCreated, http.StatusAccepted:
 		fmt.Println("Removing old function.")
 	case http.StatusNotFound:
-		fmt.Println("No existing function to remove")
+		err = fmt.Errorf("No existing function to remove")
+		fmt.Println(err.Error())
```

Comment:
@viveksyngh  when i remove the print statements, the unit tests fail. It seems the unit tests does not read the output from fmt.Errorf. Spent sometime trying to figure it out but it didn't work. Any ideas on how to do this without printing duplicate statements?

Reply:
@itohan-uk we can fix the test. Right now those tests are capturing output from stdout. Now that we are already returning error from `DeleteFunction` we can use those error value in the tests to compare.

Reply:
@viveksyngh I tried adding a function to capture from stderr but the test still fails. The build fails if the test fails. Here is the function i added
`func CaptureStderr(f func()) string {
        stdErr := os.Stderr
        r, w, _ := os.Pipe()
        defer r.Close()
        os.Stderr = w

        f()

        w.Close()
        os.Stderr = stdErr

        var b bytes.Buffer
        io.Copy(&b, r)

        return b.String()
}`

Reply:
You don't need to read values from `stdout` or `stderr`. Since `DeleteFunction` returns an error, you use that error value like `err.Error()` to compare inside unit tests. You just need to update the unit tests to remove the piece which reads value from stdout. 

I am sorry, I was not able to explain in my previous comment.

Reply:
@viveksyngh I have submitted a new patch and fixed the unit tests.